ldap/ldap.go

```diff
@@ -168,22 +186,75 @@ func (u *userService) isMemberOf(user, group string) error {
 			0,
 			0,
 			false,
-			fmt.Sprintf("(uid=%s)", user),
+			fmt.Sprintf("(uid=%s)", username),
 			[]string{"ismemberof"},
 			nil))
 	if err != nil {
 		return errors.Wrap(err, "problem searching ldap")
 	}
 	if len(result.Entries) == 0 {
-		return errors.Errorf("no entry returned for user '%s'", user)
+		return errors.Errorf("no entry returned for user '%s'", username)
 	}
 	if len(result.Entries[0].Attributes) == 0 {
-		return errors.Errorf("entry's attributes empty for user '%s'", user)
+		return errors.Errorf("entry's attributes empty for user '%s'", username)
 	}
 	for i := range result.Entries[0].Attributes[0].Values {
-		if result.Entries[0].Attributes[0].Values[i] == group {
+		if result.Entries[0].Attributes[0].Values[i] == u.Group {
 			return nil
 		}
 	}
-	return errors.Errorf("user '%s' is not a member of group '%s'", user, group)
+	return errors.Errorf("user '%s' is not a member of group '%s'", username, u.Group)
 }
+
+func (u *userService) getUser(username string) (gimlet.User, error) {
+	result, err := u.conn.Search(
+		ldap.NewSearchRequest(
+			u.Path,
+			ldap.ScopeWholeSubtree,
+			ldap.NeverDerefAliases,
+			0,
+			0,
+			false,
+			fmt.Sprintf("(uid=%s)", username),
+			[]string{},
+			nil))
+	if err != nil {
+		return nil, errors.Wrap(err, "problem searching ldap")
+	}
+	if len(result.Entries) == 0 {
+		return nil, errors.Errorf("no entry returned for user '%s'", username)
+	}
+	if len(result.Entries[0].Attributes) == 0 {
+		return nil, errors.Errorf("entry's attributes empty for user '%s'", username)
+	}
+	return makeUser(result), nil
+}
+
+func makeUser(result *ldap.SearchResult) gimlet.User {
+	user := &User{}
+	for _, entry := range result.Entries[0].Attributes {
+		if entry.Name == "uid" {
+			user.ID = entry.Values[0]
+		}
+		if entry.Name == "cn" {
+			user.Name = entry.Values[0]
+		}
+		if entry.Name == "mail" {
+			user.EmailAddress = entry.Values[0]
+		}
+	}
+	return user
+}
+
+type User struct {
```

Comment:
There's a basic user in the gimlet package that you should be able to use without needing to do thise.

Reply:
Done.